src/HaskellCI/GitHub.hs

```diff
@@ -536,6 +536,15 @@ makeGitHub _argv config@Config {..} gitconfig prj jobs@JobVersions {..} = do
             -- disable-documentation disables docs in deps: https://github.com/haskell/cabal/issues/7462
             sh_if range $ "$CABAL v2-haddock --disable-documentation" ++ haddockFlags ++ " $ARG_COMPILER --with-haddock $HADDOCK $ARG_TESTS $ARG_BENCH all"
 
+        when runWeeder $
+            let range = RangeGHC /\ Range cfgWeeder in
+            let ifCond = ghCompilerVersionArithPredicate allVersions range in
+            for_ pkgs $ \Pkg{pkgName} -> do
+                githubUsesIf "weeder" "freckle/weeder-action@v2" ifCond $ buildList $ do
+                    item ("ghc-version", ghWrapExpr "matrix.compilerVersion")
```

Comment:
I don't see this working. Comparing strings like `8.10.7` and `8.8.4` with operators like `>=` wouldn't work as intended.

---

Add test

Reply:
> Comparing strings like `8.10.7` and `8.8.4` with operators like `>=` wouldn't work as intended.

The strings get converted to numeric representation before comparison.  E.g., with [this config](https://github.com/swarm-game/swarm/blob/53b8a3b7ad2a1a874e17fea087cf6ef30957fa53/.github/workflows/cabal.haskell-ci#L16):
```
weeder: >= 9.4 && < 9.8
```
the [resulting comparison](https://github.com/swarm-game/swarm/blob/53b8a3b7ad2a1a874e17fea087cf6ef30957fa53/.github/workflows/haskell-ci.yml#L240) is:
```
${{ env.HCNUMVER >= 90400 && env.HCNUMVER < 90800 }}
```
And here's a [live demonstration](https://github.com/swarm-game/swarm/pull/2060#discussion_r1684957641) featuring coomparison against a number like `9.10.7`.

I have also now added a golden test.